### DISCUSSION TITLE
Pop up window for each cacheView

### DISCUSSION THREAD

**Discussion by Y1annis** [mortbopet/Ripes#267] [General]
Greetings

I am trying to add a way to be able to view both instruction cache and data cache at the same time. For that I am trying to implement a pop up window for each cacheView/cacheWidget. I wanted to ask if that is possible with the current implementation of Ripes.

Thank you for your time.

**Comment by mortbopet** [comment]
You might be able to place the cache widgets inside a [QDockWidget](https://doc.qt.io/qt-6/qdockwidget.html) which would allow them to either be locked into the main window or popped out. If that doesn't work (if QDockWidget strictly needs to have a `QMainWindow` parent) then you could e.g. add a pop-out button to the cache widget which would remove it from its existing layout, and instead create a new, non-modal QDialog and set the inner widget of that dialog to the cache widget.

**Reply by Y1annis** [reply]
Sorry for the late reply. I tried doing it with a QDockWidget but when i pop out the dockwidget I can't get it to dock back.

Thank you for your time.
